### PR TITLE
feat: add emulator detection property to static context

### PR DESCRIPTION
**Pull request by thisames**
## Description

This PR adds emulator detection to the static context and updates the tests to verify this functionality.

- Implemented `isEmulator` method to detect if the app is running on an emulator.
- Integrated emulator detection into the static context cache.
- Added assertions to verify emulator detection in the static context.
- Ensured static context includes the new `\$is_emulator` property.

## Motivation and Context

A crucial property for future use as a filter.

#113 

## How Has This Been Tested?

The tests have been updated to include checks for the new `\$is_emulator` property in the static context.
I tested it manually on both an emulator and a physical device.

## Checklist

- [x] I have reviewed the submitted code.
- [x] I have added tests to verify the changes.
- [ ] I have updated the documentation if necessary.
- [x] No breaking changes or changes to the changelog entry.

### CODE REVIEW

File: posthog-android/src/main/java/com/posthog/android/internal/PostHogAndroidContext.kt [hunk 1]
```diff
@@ -170,4 +172,20 @@ internal class PostHogAndroidContext(
 
         return dynamicContext
     }
+
```
Comment (thisames) [inline posthog-android/src/main/java/com/posthog/android/internal/PostHogAndroidContext.kt:175]: I was unsure whether to include the reference/inspiration, but the link is huge, and it looked odd in the code.

Comment (marandaneto) [reply posthog-android/src/main/java/com/posthog/android/internal/PostHogAndroidContext.kt:175]: Just add as a comment on top of the var

**Comment by marandaneto** [comment]
@thisames please add a changelog entry =)

**Comment by marandaneto** [comment]
@thisames would you be willing to add to our taxonomy on the backend/frontend as well? See example https://github.com/PostHog/posthog/blob/46e6624cd9b314c67cd1f076f41eda901cd9b1f6/frontend/src/lib/taxonomy.tsx#L844-L848
Its as simple as the linked lines

**Comment by thisames** [comment]
> @thisames please add a changelog entry =)

lets waiting #153 be merged? to avoid future conflicts

**Comment by thisames** [comment]
> @thisames would you be willing to add to our taxonomy on the backend/frontend as well? See example https://github.com/PostHog/posthog/blob/46e6624cd9b314c67cd1f076f41eda901cd9b1f6/frontend/src/lib/taxonomy.tsx#L844-L848
> Its as simple as the linked lines

sure!

**Comment by marandaneto** [comment]
> > @thisames please add a changelog entry =)
> 
> lets waiting #153 be merged? to avoid future conflicts

done

**Review by marandaneto — APPROVED** [review]

**Pull request merged**